pkg/filebacked/file.go

```diff
@@ -348,18 +405,28 @@ func (r *Reader) open() (err error) {
 		err = liberr.Wrap(err)
 		return
 	}
+	log.V(1).Info(
+		"reader: opened.",
+		"path",
+		r.path)
 
 	return
 }
 
 //
 // Close the reader.
 func (r *Reader) Close() {
-	if r.file != nil {
-		_ = r.file.Close()
+	defer func() {
 		_ = os.Remove(r.path)
```

Comment:
Make sure the hard-link is deleted even if the reader (iterator) is never read.  Eg: Next() is never called.